e2e/allergy/allergy-crud.e2e-spec.ts

```diff
@@ -0,0 +1,150 @@
+            loginPage.appVersion, 'userName');
+    });
+
+    function getFormData(i?: number): FormData[]{
+        const baseAllergyValues = ['Name', 'A sign', 'A symptom'];
+        const empty = (i === undefined)
+        const formData: FormData[] = [{
+            field: allergyDetailPage.getNameInput(),
+            name: 'Name',
+            value: empty ? '' : 'Try #' + i + ': ' + baseAllergyValues[0]
+        }, {
+            field: allergyDetailPage.getSignsInput(),
+            name: 'Signs',
+            value: empty ? '' : 'Try #' + i + ': ' + baseAllergyValues[1]
+        }, {
+            field: allergyDetailPage.getSymptomsInput(),
+            name: 'Symptoms',
+            value: empty ? '' : 'Try #' + i + ': ' + baseAllergyValues[2]
+        }];
+        return (formData);
+    }
+
+    it('Open allergy creation dialog', () => {
+        const title = 'Create Allergy';
+        const buttons: ButtonState[] = [{
+            name: 'Create',
+            exist: true,
+            enable: true
+        }];
+
+        mainPage.checkPresentAndDisplayed();
+        mainPage.getAllergyAddButton()
+            .click();
+        allergyDetailPage.showNewPageAndCheckTitleAndButtons(title, buttons);
+        TestUtil.checkForm(getFormData(), 'Allergy Creation is empty');
+    });
+
+    it('Close the dialog', () => {
+        allergyDetailPage.getButtonClose()
+            .click();
+        allure.createStep('Dialog is closed', () => {
+            mainPage.checkPresentAndDisplayed();
+        })();
+    });
+    it('Create Allergies', () => {
+        for (let i = 1; i <= browser.params.repeatabilityNumberPasses; i++) {
+
+            allure.createStep('Action: Create the allergy ' + i, () => {
+
+                mainPage.getAllergyAddButton()
+                    .click();
+                allergyDetailPage.checkPresentAndDisplayed();
+
+                FormService.fillForm(getFormData(i), 'Allergy Creation Form');
+                TestUtil.checkForm(getFormData(i), 'Allergy Creation is correct');
+
+                allergyDetailPage.getButtonSubmit()
+                    .click();
+                mainPage.checkPresentAndDisplayed();
+                TestUtil.checkCount(GridService.getGridInnerComponent(mainPage.getAllergyGrid()), 'Number of Allergies', i);
+
+                GridService.getRow(mainPage.getAllergyGrid(), i - 1)
+                    .then(function(cellValues) {
+                        TestUtil.checkText(cellValues[1], 'Col Name', getFormData(i)[0].value);
+                        TestUtil.checkText(cellValues[2], 'Col Signs', getFormData(i)[1].value);
+                        TestUtil.checkText(cellValues[3], 'Col Symptoms', getFormData(i)[2].value);
+                    });
+            })();
+        }
+    });
+
+    it('Contextual menu at the allergies grid', () => {
+        const menuItems = ['Update', 'Delete'];
+        for (let row = 0; row < browser.params.repeatabilityNumberPasses; row++) {
```

Comment:
For arrays iteration is cleaner to use this syntax whenever is possible (depend on the object you're iterating is a pure array):
someArray.forEach((item) => {
});

someArray.forEach((item, index) => {
});

Reply:
In this case I think it's not an array iteration, because the limits of the iteration are imposed by a parameter which is outside the grid we are trying to process. Anyway, the code that is checking the correctness of the menu items does use the code structure you propose in the comment:  
options.forEach((option) => {
            expect(field.getText())
                .toContain(option);